src/renderer/components/+preferences/kubeconfig-syncs.tsx

```diff
@@ -0,0 +1,178 @@
+}
+
+interface Value {
+  data: KubeconfigSyncValue;
+  info: SyncInfo;
+}
+
+async function getMapEntry({ filePath, ...data}: KubeconfigSyncEntry): Promise<[string, Value]> {
+  try {
+    // stat follows the stat(2) linux syscall spec, namely it follows symlinks
+    const stats = await fse.stat(filePath);
+
+    if (stats.isFile()) {
+      return [filePath, { info: { type: "file" }, data }];
+    }
+
+    if (stats.isDirectory()) {
+      return [filePath, { info: { type: "folder" }, data }];
+    }
+
+    logger.warn("[KubeconfigSyncs]: unknown stat entry", { stats });
+
+    return [filePath, { info: { type: "unknown" }, data }];
+  } catch (error) {
+    logger.warn(`[KubeconfigSyncs]: failed to stat entry: ${error}`, { error });
+
+    return [filePath, { info: { type: "unknown" }, data }];
+  }
+}
+
+@observer
+export class KubeconfigSyncs extends React.Component {
+  syncs = observable.map<string, Value>();
+  @observable loaded = false;
+
+  async componentDidMount() {
+    const mapEntries = await Promise.all(
+      iter.map(
+        UserStore.getInstance().syncKubeconfigEntries,
+        ([filePath, ...value]) => getMapEntry({ filePath, ...value }),
+      ),
+    );
+
+    this.syncs.replace(mapEntries);
+    this.loaded = true;
+
+    disposeOnUnmount(this, [
+      reaction(() => Array.from(this.syncs.entries(), ([filePath, { data }]) => [filePath, data]), syncs => {
+        UserStore.getInstance().syncKubeconfigEntries.replace(syncs);
+      })
+    ]);
+  }
+
+  @computed get syncsList(): Entry[] | undefined {
+    if (!this.loaded) {
+      return undefined;
+    }
+
+    return Array.from(this.syncs.entries(), ([filePath, value]) => ({ filePath, ...value }));
+  }
+
+  @action
+  openFileDialog = async () => {
+    const { dialog, BrowserWindow } = remote;
+    const { canceled, filePaths } = await dialog.showOpenDialog(BrowserWindow.getFocusedWindow(), {
+      properties: ["openFile", "showHiddenFiles", "multiSelections", "openDirectory"],
+      message: "Select kubeconfig file(s) and folder(s)",
+      buttonLabel: "Sync",
+    });
+
+    if (canceled) {
+      return;
+    }
+
+    const newEntries = await Promise.all(filePaths.map(filePath => getMapEntry({ filePath })));
+
+    for (const [filePath, info] of newEntries) {
+      this.syncs.set(filePath, info);
+    }
+  };
+
+  renderEntryIcon(entry: Entry) {
+    switch (entry.info.type) {
+      case "file":
+        return <Description />;
+      case "folder":
+        return <Folder />;
+      case "unknown":
+        return <HelpOutline />;
+    }
+  }
+
+  renderEntry = (entry: Entry) => {
+    return (
+      <Paper className="entry" key={entry.filePath} elevation={3}>
+        <ListItem>
+          <ListItemAvatar>
+            <Avatar>
+              {this.renderEntryIcon(entry)}
+            </Avatar>
+          </ListItemAvatar>
+          <ListItemText
+            primary={entry.filePath}
+            className="description"
+          />
```

Comment:
I think the filePath text is too small
<img width="692" alt="Screen Shot 2021-04-28 at 7 25 15 PM" src="https://user-images.githubusercontent.com/40840436/116484704-a2b00d80-a857-11eb-9eaf-7509c39a0b7f.png">

Reply:
I agree, will fix this.

Reply:
Could the folder icon be larger too, to fill more of the circular background?